### PR TITLE
refactor: use `blob.Namespace()` from celestia-core

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2071

### CODE REVIEW

File: pkg/square/builder.go [hunk 1]
```diff
@@ -366,11 +366,6 @@ func (b *Builder) IsEmpty() bool {
 	return b.TxCounter.Size() == 0 && b.PfbCounter.Size() == 0
 }
 
-// TODO: celestia-core should provide this method for `Blob`s
```
Comment (SweeXordious) [inline pkg/square/builder.go:369]: [Question][Almost unrelated]
Why does core need to know about blobs and namespaces?

Comment (rootulp) [reply pkg/square/builder.go:369]: hmm good question. Basic types like `Blob` and `BlobTx` are defined in celestia-core 

e.g. https://github.com/celestiaorg/celestia-core/blob/d5cd08c4e13a66acc087eda5d43109928e630846/types/block.go#L1039-L1055

So at least for now celestia-core is aware of blobs

Comment (SweeXordious) [reply pkg/square/builder.go:369]: Thanks. Just checked, apparently, that's only used for tests

Comment (rootulp) [reply pkg/square/builder.go:369]: We added `Namespace()` as a method to `Blob` in celestia-core as a convenience. Perhaps you're missing this context: https://github.com/celestiaorg/celestia-core/pull/1030

We can create a separate issue to explore moving types like `Blob` and `BlobTx` to celestia-app but TBH I don't expect it to be possible without a large refactor.

Comment (SweeXordious) [reply pkg/square/builder.go:369]: Aaah just found where it's needed, thanks :+1:

**Review by SweeXordious — APPROVED** [review]

File: test/util/malicious/out_of_order_builder.go [hunk 2]
```diff
@@ -72,14 +72,14 @@ func OutOfOrderExport(b *square.Builder) (square.Square, error) {
 	// sort the blobs in order of namespace. We use slice stable here to respect the
 	// order of multiple blobs within a namespace as per the priority of the PFB
 	sort.SliceStable(b.Blobs, func(i, j int) bool {
-		return bytes.Compare(square.FullNamespace(b.Blobs[i].Blob), square.FullNamespace(b.Blobs[j].Blob)) < 0
+		return bytes.Compare(b.Blobs[i].Blob.Namespace(), b.Blobs[j].Blob.Namespace()) < 0
```
Comment (staheri14) [inline test/util/malicious/out_of_order_builder.go:75]: LGTM! what is the motivation behind this change? could not find it in the linked issue. Thanks.

**Review by staheri14 — APPROVED** [review]
